src/backend/commands/tablecmds.c

```diff
@@ -897,6 +897,17 @@ DefineRelation(CreateStmt *stmt, char relkind, Oid ownerId, char relstorage, boo
 					 errhint("Use OIDS=FALSE.")));
 	}
 
+	if (gp_add_partition_inherits_table_setting && stmt->is_add_part)
+	{
+		ListCell *lc;
+		foreach(lc, stmt->options)
+		{
+			DefElem *de = lfirst(lc);
+			if (pg_strcasecmp(de->defname, "appendonly") == 0 && pg_strcasecmp(defGetString(de), "none") != 0)
+				useChangedOpts = false;
```

Comment:
May be it will be better to add the inherits_table_setting option to pass it into validate_and_refill_options(), because validation is disabled when useChangedOpts is false.

Reply:
added